### ISSUE TITLE
expand(_all_) fails with facets

### ISSUE THREAD

**Issue by RafaARV** [dgraph-io/dgraph#1650]
expand(_all_)@facets fails when a new attribute is set in a related node.
Version/Branch 0.8.3 working from web GUI.

Steps to reproduce:

SET SCHEMA:
```
mutation {
    schema {
        group: string @index(exact,term) .
        name: string @index(exact,term) .
        edge: uid @count .
        r_edge: uid @count .
        other: string @index(exact,term) .
    }
}
```

ADD ROOT NODE:
```
mutation {
  set {
    _:root <group> "test" .
    _:root <name> "root" .
    _:root <other> "root" .
  }
}
```

ADD FIRST NODES:
```
{
  root as var(func:eq(name, "root"))@filter(eq(group,"test") and eq(name,"root"))
}

mutation{
  set{
    uid(root) <link> _:nodo .
    _:nodo <name> "nodo1" .
    _:nodo <group> "test" .
  }
}
```

```
{
  root as var(func:eq(name, "root"))@filter(eq(group,"test") and eq(name,"root"))
}

mutation{
  set{
    uid(root) <link> _:nodo .
    _:nodo <name> "nodo2" .
    _:nodo <group> "test" .
  }
}
```

```
{
  root as var(func:eq(name, "root"))@filter(eq(group,"test") and eq(name,"root"))
}

mutation{
  set{
    uid(root) <link> _:nodo .
    _:nodo <name> "nodo3" .
    _:nodo <group> "test" .
  }
}
```

ADD NODES WITH FACETS:
```
{
  root as var(func:eq(name, "nodo2"))@filter(eq(group,"test") and eq(name,"nodo2"))
}

mutation{
  set{
    uid(root) <link> _:nodo (facet_test = "2-2_1", facet_test2 = 21, facet_test3 = true).
    _:nodo <name> "nodo2_1" .
    _:nodo <group> "test" .
  }
}
```

```
{
  root as var(func:eq(name, "nodo2"))@filter(eq(group,"test") and eq(name,"nodo2"))
}

mutation{
  set{
    uid(root) <link> _:nodo (facet_test = "2-2_2", facet_test2 = 22, facet_test3 = true).
    _:nodo <name> "nodo2_2" .
    _:nodo <group> "test" .
  }
}
```

QUERY ALL GROUP "TEST" NODES **WORKS**:
```
{
    nodes as var(func:eq(group, "test"))

    full_graph(func:uid(nodes)) {
        _uid_
        expand(_all_)@facets{
            _uid_
        }
    }
}
```

SET A NEW ATTRIBUTE:
```
{
  root as var(func:eq(name, "nodo2"))@filter(eq(group,"test") and eq(name,"nodo2"))
}

mutation{
  set{
    uid(root) <other> "other_val".
  }
}
```

QUERY ALL GROUP "TEST" NODES **FAILS!!!**:
```
{
    nodes as var(func:eq(group, "test"))

    full_graph(func:uid(nodes)) {
        _uid_
        expand(_all_)@facets{
            _uid_
        }
    }
}
```

Error in GUI: NetworkError when attempting to fetch resource.: Could not connect to the server
JSON is empty.

**Comment by peterstace** [comment]
Thank you for the detailed report. I've been able to reproduce the bug and will start working on a fix.

**Comment by peterstace** [comment]
Minimum triggering example:

```
mutation {
  set {
    <0x1> <dummy> "dummy" .
    <0x2> <other> "other" .
    <0x2> <link> <0x3> (k = "v").
  }
}
{
  q(func: uid(0x1, 0x2)) {
    expand(_all_) @facets { }
  }
}
```

PR that fixes the issue: #1651

**Comment by peterstace** [comment]
Merged into `release/v0.8.3` branch at  40175d0.

Cherry-picked to master at  a42bdbe5.

**Comment by peterstace** [comment]
@RafaARV - new binaries for v0.8.3 will be released soon (available via github download or docker).
